### ISSUE TITLE
Inline asm: `*const [type error]` does not implement the Copy trait

### ISSUE THREAD

**Issue by Kyuuhachi** [rust-lang/rust#113788]
### Code

```Rust
fn main() {
	let peb: *const PEB;
	unsafe { std::arch::asm!("mov {0}, fs:[0x30]", out(reg) peb); }
}
```

### Current output

```Shell
error[E0412]: cannot find type `PEB` in this scope
 --> src/main.rs:2:18
  |
2 |     let peb: *const PEB;
  |                     ^^^ not found in this scope

error: arguments for inline assembly must be copyable
 --> src/main.rs:3:49
  |
3 |     std::arch::asm!("mov {0}, fs:[0x30]", out(reg) peb);
  |                                                    ^^^
  |
  = note: `*const [type error]` does not implement the Copy trait

For more information about this error, try `rustc --explain E0412`.
```

### Desired output

```Shell
Probably just omit the second error; no `Copy` error is given when moving an unknown type normally.
```

### Anything else?

Happens on `1.73.0-nightly (2023-07-16 0e8e857b11f60a785aea)`, as well as several earlier versions.

**Comment by compiler-errors** [comment]
Just needs someone to suppress this error if the type [`references_error`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/visit/trait.TypeVisitableExt.html#method.references_error), I think.

**Comment by Rajveer100** [comment]
@Kyuuhachi @compiler-errors 
Hi! I have given a brief introduction of myself in the `new members` [zulip channel](https://rust-lang.zulipchat.com/#narrow/stream/122652-new-members/topic/Hello!/near/383589361).

With that said, I have potentially setup the necessary pre-requisites and compiler infrastructure in my local machine, also re confirmed by using the `./x check` and other build commands which seem to work. As per the `docs`, I have the all the toolchains command executed (for `stages` [0-2]). 

Like any other contribution, I figured that in `rust-lang` repo, `good-first-issues` are tagged as `E-Easy` for new-comers, and to start, I have chosen this issue. 

Until now, I have found the following `snippet` which I presume is targeting the output:

```Rust
// Check that the type implements Copy. The only case where this can
// possibly fail is for SIMD types which don't #[derive(Copy)].
if !ty.is_copy_modulo_regions(self.tcx, self.param_env) {
    let msg = "arguments for inline assembly must be copyable";
    let mut err = self.tcx.sess.struct_span_err(expr.span, msg);
    err.note(format!("`{ty}` does not implement the Copy trait"));
    err.emit();
}
```

Would be great if one could point out any points I may have missed out! 
A good head start would be great!
